### DISCUSSION TITLE
Apache 5.0 side effect: Response objects have to be closed

### DISCUSSION THREAD

**Discussion by natan-abolafya** [dropwizard/dropwizard#7109] [Q&A]
We have lots of tests that use `DropwizardAppExtension.client()` object and makes requests that returns a `Response` object.

Now I'm upgrading to dropwizard 3.0 and tests are freezing because the not closing Response objects now keep the connections up. The connection pool fills up quick and a request freezes completely, waiting a connection from the pool.

This is a warning Intellij has been showing for quite a long while, so I guess it's on us. But at this point, changing all the tests is going to be very tedious job. So if you anybody knows how to go around it I'd appreciate it :).

**Comment by zUniQueX** [comment]
Hi @natan-abolafya. You can take a look at the `Apache5ConnectionClosingStrategy`. With an implementation of this interface you can specify the closing behaviour to be applied once the response entity has been read.

**Reply by natan-abolafya** [reply]
Hi. 
The problem is that our tests do not read the entity often. Some only check the status code for example. Then goes on to the next test or next call.

The change is mentioned here I think: https://www.baeldung.com/apache-httpclient-vs-closeablehttpclient
> The underlying HTTP connection is held by the response object to allow the response content to be streamed directly from the network socket.

But I am not sure how to address that. I guess it could be the `Apache5ConnectionClosingStrategy` but not sure I get it. The interface looks like it's about how to handle the close call, not actually triggering the close call automatically.
![image](https://github.com/dropwizard/dropwizard/assets/64091305/77f3e71b-d85b-4071-b5dc-8dd7a3f5203a)

**Reply by zUniQueX** [reply]
According to the [Jersey docs](https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest/filters-and-interceptors.html) on filters and interceptors the `ClientResponseFilter`s will be executed after retrieving the response and before returning it. With the context objects you can manipulate the entity stream there.
